ui/src/app/views/dataset/component.tsx

```diff
@@ -128,39 +137,57 @@ export const DataSet = withCurrentWorkspace()(class extends React.Component<
     const {workspaceNamespace, workspaceFirecloudName} = resource;
     const {id} = updatedResource;
     let call;
+    let updatedConceptSetList = this.state.conceptSetList;
+    let updatedCohortList = this.state.cohortList;
     if (resource.cohort) {
       call = cohortsApi().updateCohort(workspaceNamespace, workspaceFirecloudName,
         id, updatedResource as Cohort);
+      updatedCohortList = this.state.cohortList.map((cohort) => {
+        return updatedResource.id === cohort.id ? (updatedResource as Cohort) : cohort;
+      });
     } else if (resource.conceptSet) {
       call = conceptSetsApi().updateConceptSet(workspaceNamespace, workspaceFirecloudName,
         id, updatedResource);
+      updatedConceptSetList = this.state.conceptSetList.map((conceptSet) => {
+        return updatedResource.id === conceptSet.id ? (updatedResource as ConceptSet) : conceptSet;
+      });
     }
-    const edited = this.state.resource.conceptSet;
-    const updatedList = this.state.conceptSetList.map((conceptSet) => {
-      return edited.id === conceptSet.id ? edited : conceptSet;
-    });
-    this.setState({conceptSetList: updatedList, resource: undefined, rType: undefined});
+    this.setState({conceptSetList: updatedConceptSetList, cohortList: updatedCohortList,
+      resource: undefined, rType: undefined});
     call.then(() => this.closeEditModal());
   }
 
   closeEditModal(): void {
     this.setState({editing: false});
   }
 
+  clone(cohort: Cohort): void {
+    const {namespace, id} = this.props.workspace;
+    navigateByUrl(`/workspaces/${namespace}/${id}/cohorts/build?cohortId=${cohort.id}`);
+  }
```

Comment:
TOL: This seems like a strange thing to have to put here. It might be worth trying to disentangle this polymorphic 'resource' notion a bit at some point.